src/devices/machine/z80scc.cpp

```diff
@@ -2857,3 +2863,32 @@ WRITE_LINE_MEMBER(z80scc_channel::write_rx)
 	if(m_rxc != 0 || m_brg_rate != 0)
 		device_serial_interface::rx_w(state);
 }
+
+/*
+ * This is a partial implementation of the "wait/dma request" functionality of the SCC controlled by
+ * bits D7, D6 and D5 in WR1. This implementation is sufficient to support DMA request on transmit
+ * used by the InterPro driver.
+ *
+ * TODO:
+ *  - wait function (D6=0)
+ *  - wait/request function on receive (D5=1)
+ */
+void z80scc_channel::check_waitrequest()
+{
+	// don't do anything if wait/request function is not enabled
+	if ((m_wr1 & WR1_WRDY_ENABLE) == 0)
+		return;
+
+	// wait/request function for receive not implemented
+	if (m_wr1 & WR1_WRDY_ON_RX_TX)
+		return;
+
+	// if dma request function is enabled
+	if (m_wr1 & WR1_WRDY_FUNCTION)
+	{
+		// assert /W//REQ if transmit buffer is empty, clear if it's not
+		int state = (m_rr0 & RR0_TX_BUFFER_EMPTY) ? ASSERT_LINE : CLEAR_LINE;
+
+		m_index ? m_uart->m_out_wrdyb_cb(state) : m_uart->m_out_wrdya_cb(state);
```

Comment:
These are both write line devcb, aren't they?  Remember C++ `?:` operator can yield an lvalue - you can probably simplify this line slightly to:
```
(m_index ? m_uart->m_out_wrdyb_cb : m_uart->m_out_wrdya_cb)(state);
```
It's not much simpler, but it only says `state` once.